### PR TITLE
deployment

### PR DESCRIPTION
**Pull request by tkellen**
This changes a bunch of things around how secrets are managed and deployed.

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -8,10 +8,7 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
 
   # This directory should be the same as the site_path setting in
   # deploy/ansible/group_vars/all.yml
-  config.vm.synced_folder '.', '/mnt/site',
-    owner: "www-data",
```
Comment (tbranyen) [inline Vagrantfile]: Without this added the permissions were set to user: vagrant, group: vagrant which causes permission issues as the process runs on www-data.

Comment (tkellen) [inline Vagrantfile]: Hrmm. I'm not seeing this? I can start the service without those perms in place.

Comment (tbranyen) [inline Vagrantfile]: Can you confirm the user that owns the project is www-data and not vagrant?

File: deploy/deploy-staging.sh [hunk 2]
```diff
@@ -1,64 +0,0 @@
-#!/usr/bin/env bash
```
Comment (tbranyen) [inline deploy/deploy-staging.sh]: Did you mean to delete this file? It's referenced in package.json to deploy

Comment (tkellen) [inline deploy/deploy-staging.sh]: I only see entries for deploy-production and deploy-staging?

File: Vagrantfile (hunk 1 above)
Comment (tkellen) [inline Vagrantfile]: The whole vagrant folder is owned by `vagrant:vagrant`, but i can do `sudo -u www-data npm start` from /mnt/site without issue.

Comment (tbranyen) [inline Vagrantfile]: Okay, sorry for being a pest with this, it's just that I had this be an issue before (when I was creating the log file).  I'm not sure if it'll come up again.  If it does, wanted to make sure there was a comment trail around the change.

Comment (tkellen) [inline Vagrantfile]: Roger that, no worries!

File: deploy/deploy-staging.sh (hunk 2 above)
Comment (tbranyen) [inline deploy/deploy-staging.sh]: OH, I'm sorry, I totally read the GitHub diffs wrong.  I noticed deploy-production was deleted, but it looks like it was changed. O_o

Comment (tkellen) [inline deploy/deploy-staging.sh]: Ah yeah, @cowboy's got some fancy bash scripting going on that makes it work based on the filename so you can just symlink additional ones.  Honestly, I'm on the fence about if the bash scripts are a good idea--even though it shows the Ansible command being run I feel like it'd be better if people just used Ansible directly.

Comment (tbranyen) [inline deploy/deploy-staging.sh]: Am I reading the diff right such that provision and deploy are the same command now? Deploy is just an alias for provision? If that's the case we can consolidate down the package.json to just `npm run deploy` right?

Comment (tkellen) [inline deploy/deploy-staging.sh]: Deploy is a symlink and the filename determines which playbook and inventory file to use.

**Comment by tbranyen** [comment]
Thanks! :grin:

**Comment by tbranyen** [comment]
FWIW I now have to run:

``` sh
npm run deploy -- --ask-sudo-pass
```

Am I doing something wrong, or is that how it should be done?

**Comment by tbranyen** [comment]
Also this just happened when I tried to run that deploy script:

```
TASK: [deploy | download secret files from secret server] ********************* 
failed: [email.bocoup.com -> 127.0.0.1] => (item=ses.json) => {"changed": true, "cmd": "scp -3 -r tim@nest.bocoup.com:/mnt/secrets/ses.json ~/ses.json", "delta": "0:00:00.320546", "end": "2015-05-08 12:33:36.009245", "item": "ses.json", "rc": 1, "start": "2015-05-08 12:33:35.688699", "warnings": []}
stderr: @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
@    WARNING: REMOTE HOST IDENTIFICATION HAS CHANGED!     @
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
IT IS POSSIBLE THAT SOMEONE IS DOING SOMETHING NASTY!
Someone could be eavesdropping on you right now (man-in-the-middle attack)!
It is also possible that a host key has just been changed.
```

**Comment by tbranyen** [comment]
Okay clearing out my known_hosts fixed that issue.

**Comment by tbranyen** [comment]
Now this:

```
TASK: [deploy | copy secret files to correct location] ************************ 
failed: [email.bocoup.com] => (item=ses.json) => {"changed": true, "cmd": ["mv", "~/ses.json", "/mnt/site/config/"], "delta": "0:00:00.001925", "end": "2015-05-08 16:35:26.250406", "item": "ses.json", "rc": 1, "start": "2015-05-08 16:35:26.248481", "warnings": []}
stderr: mv: cannot stat ‘/root/ses.json’: No such file or directory

FATAL: all hosts have already failed -- aborting
```

I'll do some debugging.

**Comment by tkellen** [comment]
RE: ask_sudo_pass...

I'm not able to duplicate that? This should be enforced by:
https://github.com/bocoup/email/blob/master/deploy/ansible/ansible.cfg#L3

**Comment by tbranyen** [comment]
```
λ ansible --version
ansible 1.9.1
  configured module search path = None
```

Maybe it's my version? It's definitely not prompting me ATM though.

**Pull request merged**